### PR TITLE
2.x: Fix MaybeTimber by using scheduler and unit

### PR DESCRIPTION
**Pull request by vanniktech**
Also one question why does `CompletableTimer` check `if (!sd.isDisposed()) {` before replacing but SingleTimer does not? I adjusted MaybeTimber to SingleTimer so I didn't add the check there either.

However I feel like this should be consistent, can you elaborate which one is wanted?

### CODE REVIEW

File: src/main/java/io/reactivex/Maybe.java [hunk 1]
```diff
@@ -1238,7 +1238,7 @@
     }
 
     /**
-     * Returns a Flowable that emits one item after a specified delay on a specified Scheduler.
+     * Returns a Maybe that emits one item after a specified delay on a specified Scheduler.
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Maybe.java]: I did a cleanup of bunch of these so you may need to rebase this after.

File: src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java [hunk 2]
```diff
@@ -38,40 +38,16 @@ public MaybeTimer(long delay, TimeUnit unit, Scheduler scheduler) {
...
-        @Override
-        public boolean isDisposed() {
-            return DisposableHelper.isDisposed(get());
-        }
+        observer.onSubscribe(sd);
 
-        void setFuture(Disposable d) {
-            DisposableHelper.replace(this, d);
-        }
+        sd.replace(scheduler.scheduleDirect(new Runnable() {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java]: It was okay with the named class, we should be moving away from anonymous inner classes.

File: src/main/java/io/reactivex/Maybe.java (hunk 1 above)
Comment (vanniktech) [reply src/main/java/io/reactivex/Maybe.java]: This PR was created on the current 2.x so after you posted that earlier PR in any case if you cleaned this one up already I could also just revert this one change

File: src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java (hunk 2 above)
Comment (vanniktech) [reply src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java]: So should I change it back and adapt SingleTimer & CompletableTimer too?

Comment (akarnokd) [reply src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java]: Yes

Comment (vanniktech) [reply src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java]: Alright and what about the `isDisposed` check in CompletableTimber?

Comment (akarnokd) [reply src/main/java/io/reactivex/internal/operators/maybe/MaybeTimer.java]: Drop it.

File: src/main/java/io/reactivex/internal/operators/completable/CompletableTimer.java [hunk 3]
```diff
@@ -14,10 +14,17 @@
 package io.reactivex.internal.operators.completable;
 
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 
-import io.reactivex.*;
-import io.reactivex.internal.disposables.SequentialDisposable;
+import io.reactivex.Completable;
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/completable/CompletableTimer.java]: Please update your IDE settings to not unroll \* imports.

**Pull request merged**